### PR TITLE
[Overflow-400] Refactor Profile Page

### PR DESCRIPTION
**Pull request by IanUrriza10**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-400
## Commands

`php artisan storage:link` on backend Docker

## Expected Output
* Correct Styling
* Profile Card
  * Profile Data
  * Follower Modal
    * has option to follow/unfollow 
  * Following Modal
    * has option to follow/unfollow 
  * Edit Profile
    * First Name (max chars 30)
    * Last Name (max chars 30)
    * About Me (max chars 250)
    * Can upload new photo
    * Will show a preview when selecting a new photo
    * Will check for valid image types
    * Errors for required and exceeding max chars present 
    * Max file size is 5MB
* Activity
  * Top Questions
  * Top Answers 
  * Clicking on Card will direct to the question link
* Bookmarks
  * infinite scrollable 
  *  clicking on Bookmark icon removes bookmark
  * Clicking on Card will direct to the question link
  * also reloads data
## Notes
* Will update the values on the slice 
  * first_name
  * last_name
  * avatar 
## Screenshots
Edit Modal
image
image
image

Follower Modals
image
Following Modals
image

image
image

### CODE REVIEW

File: frontend/components/molecules/ProfileCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                        <div className="">
+                            <span className="font-bold">{following_count}</span> Following
+                        </div>
+                        <div className="">
+                            <span className="font-bold">{follower_count}</span> Followers
+                        </div>
+                    </div>
+                    {toggleFollow && (
+                        <div className="w-full">
+                            <button
```
Comment (nicoleamber) [inline frontend/components/molecules/ProfileCard/index.tsx]: Please use the `Button` component for uniformity.

File: frontend/components/templates/Profile.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
+import { omit } from 'lodash'
+import ProfileCard, { type ProfileCardProps } from '../molecules/ProfileCard'
+import SummaryCard from '../molecules/SummaryCard'
+import { type ITag } from '../molecules/TagsInput'
+type MITag = ITag & { isWatched: boolean }
+// const tempTags: MITag[] = [
```
Comment (nicoleamber) [inline frontend/components/templates/Profile.tsx]: Remove comment

File: frontend/components/templates/layouts/TopLeftLayout.tsx [hunk 3]
```diff
@@ -6,16 +6,17 @@ type Props = {
 }
 
 const TopLeftLayout = ({ children }: Props): JSX.Element => {
+    console.log('TopLeftLayout')
```
Comment (nicoleamber) [inline frontend/components/templates/layouts/TopLeftLayout.tsx]:
```suggestion

```

File: frontend/components/templates/Profile.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    data: ProfileData
+    toggleFollow?: (input: { variables: { userId: number } }) => void
+    isPublic: boolean
+}
+
+const ProfileLayout = ({ data, toggleFollow, isPublic }: ProfileLayoutProps): JSX.Element => {
+    const renderActivities = (): JSX.Element => {
+        return (
+            <div className="w-full space-y-4 bg-white pb-4 drop-shadow-md">
+                <div className="bg-[#F5D3CF]  p-2 font-semibold leading-6">ACTIVITY</div>
```
Comment (nicoleamber) [inline frontend/components/templates/Profile.tsx]: Please set the colors in the tailwind config file and apply changes to color classes. 

* _You can copy from here for the mean time_: https://github.com/framgia/sph-sun-overflow/pull/266/files#diff-8fdab0c1827a182dfaa61b1253086037a44105350409f6701c49ebe130f6705b

**Review by nicoleamber — CHANGES_REQUESTED** [review]

File: frontend/components/molecules/ProfileCard/index.tsx (hunk 1 above)
Comment (IanUrriza10) [reply frontend/components/molecules/ProfileCard/index.tsx]: Done

File: frontend/components/templates/Profile.tsx (hunk 2 above)
Comment (IanUrriza10) [reply frontend/components/templates/Profile.tsx]: Done

**Review by nicoleamber — DISMISSED** [review]

**Review by jules-lucero — DISMISSED** [review]

**Review by nicoleamber — DISMISSED** [review]

**Review by jules-lucero — DISMISSED** [review]

**Review by nicoleamber — DISMISSED** [review]

**Comment by nicoleamber** [comment]
## QA Feedback

* Tag in Activity Questions should follow the design in Figma
* Votes should be shown as percentage with the Like icon
* Include the date in Activity cards
* Set gap space for Bookmark cards
* Bookmark icon cannot be clicked for unbookmark action

**Edit Profile**
* Limit first name and last name to less than 30 characters
* Show validation notification when there are submission errors
* Reset edit form when there are submission errors
* Upload photo not working

[Screencast 2023-04-20 18-40-51.webm](https://user-images.githubusercontent.com/116238730/233343601-29f3670f-29ea-40d4-9046-ecd5560c6ecf.webm)

**Comment by ipilikent** [comment]
## Comment

1. Profile Card
- Width - 235px
2.  About Me Field
- There is validation if it exceeds 250 characters
    - Error Message "Maximum of 250 characters"
3. Follow Button
- Width - 187px
4. Like Icon
- Height - 13px Width - 13px

**Review by jules-lucero — DISMISSED** [review]

**Review by nicoleamber — DISMISSED** [review]

**Review by jules-lucero — DISMISSED** [review]

**Comment by ipilikent** [comment]
## Approved PR
- All displayed the same as the Figma design
# Profile Card
- The following and followers modals are displayed
   - Can unfollow user
   - Can follow user
   - Can see followers and following of user
- Can edit profile
   - First Name (max chars 30)
   - Last Name (max chars 30)
   - About Me (max chars 250)
   - Can upload new photo
   - Will show a preview when selecting a new photo
   - Can upload photos/GIF
   - Profile Stats is displayed
   - The position of user is displayed
   - The about me is displayed
# Activity Section
   - Can click top questions and top answers
   - Can click Tags
   - The top 5 questions is displayed
   - The top 5 answers displayed
# Bookmark Section
   - Can toggle bookmark
   - Can remove bookmark
   - Can click questions and answers in the bookmark
   - The bookmark section is infinite scrollable
- Responsive Profile Page

## Proof
<details>

400 - Refactor Profile Page

</details>

Test Case:  https://docs.google.com/spreadsheets/d/1s5kbSLHk9yzEaLtnf0vnQ-5NvOUIh0mDYCXciv4q0KI/edit#gid=792229086
-----

- Total Test time: 0.75 hr.
- Total Comments: 0
- Total Iterations: 0

**Review by ipilikent — APPROVED** [review]

**Review by jules-lucero — APPROVED** [review]

**Review by kenorenzsun — APPROVED** [review]

**Pull request merged**